### DISCUSSION TITLE
sh-rpi config file

### DISCUSSION THREAD

**Discussion by HPN-1956** [hatlabs/discussions#74] [Q&A] [answered]
In which folder is the config file?

**Comment by mairas** [comment]
The config file is `/etc/shrpid.conf`. It seems the info indeed isn't in the actual documentation, but at least it is mentioned in the repository README: https://github.com/hatlabs/SH-RPi-daemon

**Reply by HPN-1956** [reply]
The config file is not in the /etc folder. In fact it is not existing at all on file.
Please advise how to create the /etc/shrpid.conf file and which content is mandatory.

**Comment by mairas** [comment]
Yes, there is no file by default. I should've been clearer in my wording. I now added documentation on the config file here: https://docs.hatlabs.fi/sh-rpi/docs/software/#daemon-configuration-file

**Reply by HPN-1956** [reply, marked as answer]
Thanks for the reply.
I have now created the config file and moved my customisation from /usr/lib/systemd/system/shrpid.service to /etc/shrpid.conf. 

And it works perfectly.

The goal of my customization was to close opencpn before shutdown because opencpn can not handle being shutdown when active  . 

My modifications to shrpid.service:

![image](https://github.com/user-attachments/assets/a137ea24-7a42-4edd-949f-3b999e288632)

My modifications to shrpid.conf:

![image](https://github.com/user-attachments/assets/dce7f8fa-6283-4eab-8454-e8b03683381d)

My shutdown script:

![image](https://github.com/user-attachments/assets/5bbdd91e-98d5-4a65-b1c5-1d95f764ce57)

best regards

Hans Peter

**Comment by mairas** [comment]
There was actually a relatively severe bug in the config file handling function, and none of the config keys with dashes would be applied correctly. I have now hopefully fixed the problem and released a new version. It can be installed by running the installer script again:

```
curl -L \
    https://raw.githubusercontent.com/hatlabs/SH-RPi-daemon/main/install-online.sh \
    | sudo bash
```

**Reply by HPN-1956** [reply]
I have installed the fix and my customization still works.
Thanks again.
